src/impl/mpispace/Kokkos_MPISpace_Ops.hpp

```diff
@@ -114,18 +114,30 @@ KOKKOS_REMOTESPACES_ATOMIC_FETCH(double)
 
 #undef KOKKOS_REMOTESPACES_ATOMIC_FETCH
 
-#define KOKKOS_REMOTESPACES_ATOMIC_ADD(type)                                   \
-  static KOKKOS_INLINE_FUNCTION void mpi_type_atomic_add(type *ptr,            \
-                                                         type value, int pe) { \
-    /* TBD */                                                                  \
-  }
-
-KOKKOS_REMOTESPACES_ATOMIC_ADD(int)
-KOKKOS_REMOTESPACES_ATOMIC_ADD(unsigned int)
-KOKKOS_REMOTESPACES_ATOMIC_ADD(long)
-KOKKOS_REMOTESPACES_ATOMIC_ADD(unsigned long)
-KOKKOS_REMOTESPACES_ATOMIC_ADD(long long)
-KOKKOS_REMOTESPACES_ATOMIC_ADD(unsigned long long)
+#define KOKKOS_REMOTESPACES_ATOMIC_ADD(type, mpi_type)                     \
+  static KOKKOS_INLINE_FUNCTION void mpi_type_atomic_add(                  \
+      type &value, type &retVal, int offset, int pe, const MPI_Win &win) { \
+    assert(win != MPI_WIN_NULL);                                           \
+    int _typesize;                                                         \
+    MPI_Type_size(mpi_type, &_typesize);                                   \
```

Comment:
Why not `sizeof(type)`?

Reply:
Just copied from: https://github.com/vmiheer/kokkos-remote-spaces/blob/f385ad7dce39b9d82801d6803876366747d2d772/src/impl/mpispace/Kokkos_MPISpace_Ops.hpp#L33